### PR TITLE
Implement the User-SupportEmail Relationship

### PR DESCRIPTION
**Pull request by Henry0422**
#37, created User-SupportEmail relationship, and now the support email is associated with the specific user account.
#36, added the 'mlang' query to check if the email address existence.
#35, get back the connected-time on support email page.

### CODE REVIEW

File: src/foam/support/modal/NewEmailSupportModal.js [hunk 1]
```diff
@@ -101,12 +105,19 @@ foam.CLASS({
...
+        name: 'emailRegex',
+        value: /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
       }
     ],
 
     messages:[
       { name:'title', message:'New Email' },
       { name:'titlelabel', message:'Input the email address you want to connect to the help desk.' },
+      { name:'emailInvalid', message: 'The email you have entered is invalid, try again.'},
+      { name:'emailNotExist', message: 'The email you have entered is existed.'}
```
Comment (zahnno) [inline src/foam/support/modal/NewEmailSupportModal.js]: There's a minor typo in the message here. Replace with the following: 'The email you have entered already exists.' Also the name of this message is a little confusing. Rename it to something more suggestive of the message like: "EmailExists".

File: src/foam/support/modal/NewEmailSupportModal.js [hunk 2]
```diff
@@ -101,12 +105,19 @@ foam.CLASS({
...
+        class: 'String',
+        name: 'emailRegex',
+        value: /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
       }
     ],
 
     messages:[
       { name:'title', message:'New Email' },
       { name:'titlelabel', message:'Input the email address you want to connect to the help desk.' },
+      { name:'emailInvalid', message: 'The email you have entered is invalid, try again.'},
```
Comment (zahnno) [inline src/foam/support/modal/NewEmailSupportModal.js]: Capitilize the "t" in "try again."

File: src/foam/support/view/SupportEmailView.js [hunk 3]
```diff
@@ -5,11 +5,12 @@ foam.CLASS({
...
-    'foam.u2.dialog.Popup'
+    'foam.u2.dialog.Popup',
+    'foam.u2.view.TableView'
   ],
 
   imports: [ 
-    'supportEmailDAO', 
+    'user',
     'createLabel',  
     'ctrl'
```
Comment (zahnno) [inline src/foam/support/view/SupportEmailView.js]: Sort imports alphabetically

File: src/foam/support/view/SupportEmailView.js [hunk 4]
```diff
@@ -5,11 +5,12 @@ foam.CLASS({
 
   requires: [ 
     'foam.u2.ListCreateController',
-    'foam.u2.dialog.Popup'
+    'foam.u2.dialog.Popup',
+    'foam.u2.view.TableView'
```
Comment (zahnno) [inline src/foam/support/view/SupportEmailView.js]: Sort

File: src/foam/support/view/SupportEmailView.js [hunk 5]
```diff
@@ -119,24 +123,28 @@ foam.CLASS({
...
         self.emptyDAO = a.array.length == 0;
       });
 
       this
       .addClass(this.myClass())
       .start().addClass('Rectangle-11-Copy')
         .start().addClass('title1')
-          .start().add('Support Email Management').addClass('title').end()
+          .start()
+            .add('Support Emails Management').addClass('title')
```
Comment (zahnno) [inline src/foam/support/view/SupportEmailView.js]: Let's make a message for this.

Comment (Henry0422) [reply src/foam/support/view/SupportEmailView.js]: `.start().add('Support Email Management').addClass('title').end()` is equal to `<div class='title'>Support Email Management</div>`. Is it necessary to put this string in a message variable when we only use this string once?

If we put `Support Email Management` in a message, we may need to put `No support email connected` in the message, as well, for consistency.

File: src/foam/support/view/SupportEmailView.js (hunk 4 above)
Comment (Henry0422) [reply src/foam/support/view/SupportEmailView.js]: OK

File: src/foam/support/view/SupportEmailView.js (hunk 3 above)
Comment (Henry0422) [reply src/foam/support/view/SupportEmailView.js]: OK

File: src/foam/support/modal/NewEmailSupportModal.js (hunk 1 above)
Comment (Henry0422) [reply src/foam/support/modal/NewEmailSupportModal.js]: Sure

File: src/foam/support/view/SupportEmailView.js (hunk 5 above)
Comment (Henry0422) [reply src/foam/support/view/SupportEmailView.js]: Will update this.

**Pull request merged**